src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -109,17 +109,36 @@ public function getMaxDepth()
     /**
      * {@inheritdoc}
      */
-    public function setSerializedName(string $serializedName = null)
+    public function setSerializedNames(string $serializedName = null, array $groups = [])
```

Comment:
These method changes are BC breaks. You'll have to introduce a new methods (as done here) but deprecate the old ones instead of dropping them directly.

Reply:
just for my culture. It is a BC break even the interface is marked as internal ? If AttributeMetadata was marked as final, I could do this kinf of modifications ?

Reply:
done